src/xtgeo/cube/_cube_import.py

```diff
@@ -127,13 +157,218 @@ def import_segy(sfile: xtgeo._XTGeoFile) -> Dict:
-        "values": values,
         "yflip": yflip,
-        "segyfile": sfile,
         "traceidcodes": traceidcodes,
     }
 
 
+def _import_segy_incomplete_traces(segyfile: segyio.segy.SegyFile) -> Dict:
+    """Import a a cube SEGY with incomplete traces via the SegyIO library.
+
+    Note that the undefined value will be xtgeo.UNDEF (large number)!
+
+    It is also logical to treat missing traces as dead traces, i.e. the should
+    get value 2 in traceidcodes
+
+    Args:
+        segyfile: Filehandle from segyio
+    """
+    segyfile.mmap()
+    # get data (which will need padding later for missing traces)
+    data = segyfile.trace.raw[:]
+
+    trcode = segyio.TraceField.TraceIdentificationCode
+    traceidcodes_input = segyfile.attributes(trcode)[:]
+
+    ilines_case = np.array([h[189] for h in segyfile.header])
+    xlines_case = np.array([h[193] for h in segyfile.header])
+
+    # detect minimum inline and xline spacing (e.g.sampling could be every second)
+    idiff = np.diff(ilines_case)
+    xdiff = np.diff(xlines_case)
+    ispacing = int(np.abs(idiff[idiff != 0]).min())
+    xspacing = int(np.abs(xdiff[xdiff != 0]).min())
+
+    ncol = int(abs(ilines_case.min() - ilines_case.max()) / ispacing) + 1
+    nrow = int(abs(xlines_case.min() - xlines_case.max()) / xspacing) + 1
+    nlay = data.shape[1]
+
+    values = np.full((ncol, nrow, nlay), xtgeo.UNDEF, dtype=np.float32)
+    traceidcodes = np.full((ncol, nrow), 2, dtype=np.int64)
+
+    ilines_shifted = (ilines_case / ispacing).astype(np.int64)
+    ilines_shifted -= ilines_shifted.min()
+    xlines_shifted = (xlines_case / xspacing).astype(np.int64)
+    xlines_shifted -= xlines_shifted.min()
+
+    values[ilines_shifted, xlines_shifted, :] = data
+    values = _validate_cube_values(values)
+    traceidcodes[ilines_shifted, xlines_shifted] = traceidcodes_input
+
+    # generate new ilines, xlines vector with unique values
+    ilines = np.array(
+        range(ilines_case.min(), ilines_case.max() + ispacing, ispacing), dtype=np.int32
+    )
+    xlines = np.array(
+        range(xlines_case.min(), xlines_case.max() + xspacing, xspacing), dtype=np.int32
+    )
+
+    attrs = _geometry_incomplete_traces(
+        segyfile,
+        ncol,
+        nrow,
+        ilines,
+        xlines,
+        ilines_case,
+        xlines_case,
+        ispacing,
+        xspacing,
+    )
+
+    attrs["ncol"] = ncol
+    attrs["nrow"] = nrow
+    attrs["nlay"] = nlay
+    attrs["ilines"] = ilines
+    attrs["xlines"] = xlines
+    attrs["values"] = values
+    attrs["traceidcodes"] = traceidcodes
+    return attrs
+
+
+def _inverse_anyline_map(anylines: dict) -> Dict:
+    """Small helper function to get e.g. inline 2345: [0, 1, 2, .., 70].
+
+    I.e. to get a mapping between inline number and a list of possible indices
+
+    """
+    anyll = {}
+    for ind, key in enumerate(anylines):
+        if key not in anyll:
+            anyll[key] = list()
+        anyll[key].append(ind)
```

Comment:
You can use a `defaultdict` here:
```suggestion
    anyll = defaultdict(list)
    for ind, key in enumerate(anylines):
        anyll[key].append(ind)
```
with:
```
from collections import defaultdict
```

Reply:
Though what is the purpose here? Looping over the dict will loop over keys, which will be unique, so wont the result be just a list of one number for each key?
i.e. this equals:
```
{key: [i] for i, key in enumerate(anylines)}
```
?

Reply:
Exactly; I got it way to complicated... thanks

Reply:
> Exactly; I got it way to complicated... thanks. Edit: That did not work as you will not add to [i]. I.e. the tests was failing. I keep the defaultdict solution

but is the input not a dictionary?

```
def foo(anylines: dict) -> dict:
    anyll = defaultdict(list)
    for ind, key in enumerate(anylines):
        anyll[key].append(ind)
    return anyll
```

```
foo({"foo": 1, "bar": 2, "foo": 3})  # duplicate keys
>>> defaultdict(<class 'list'>, {'foo': [0], 'bar': [1]})
```
if it is a list:
```
foo(["foo", "bar", "foo"])
>>> defaultdict(<class 'list'>, {'foo': [0,2], 'bar': [1]})
```

```
{key: [i] for i, key in enumerate({"foo": 1, "bar": 2, "foo": 3})}
>>> {'foo': [0], 'bar': [1]}
```

edit: looks like the typing is wrong, the input is a `List[int]`, not a dict, so the function can stay as is, just needs updated typing.